spec/features/project_management_spec.rb

```diff
@@ -86,6 +86,7 @@
     visit '/projects/new'
     fill_in 'project_description', with: '_italic_ **bold**'
     click_link 'Preview'
+    wait_for_preview
```

Comment:
Hm... `have_css` already waits the `Capybara.default_wait_time` so would `page.find('#preview-contents')`. Why do we need this method? What are you trying to achieve?

Reply:
here capybara doesn't wait for our ajax request of previewing markdown to complete so it doesn't find that div.
So from this [link](https://robots.thoughtbot.com/automatically-wait-for-ajax-with-capybara), I tried to add this function but the error still persists.

Reply:
Did you follow the [first link](https://github.com/teamcapybara/capybara#asynchronous-javascript-ajax-and-friends) from your blog post? This is what the capybara README has to say:

```
When working with asynchronous JavaScript, you might come across situations
where you are attempting to interact with an element which is not yet present
on the page. Capybara automatically deals with this by waiting for elements to
appear on the page.
```

The rest of the thoughtbot blog post is about fixing race conditions between two requests happening at the same time (`click_link` & `reload_page`), this is not what is happening here.

So effectively you just have increased the time capbybara waits for the `bold` and `italic` elements to appear from 2 seconds (default) to 7 seconds. Which can also do by setting `Capybara.default_wait_time = 7` instead of adding all this code.

So why where you doing this? Did you run into a timeout locally?

Reply:
Yes, I was having timeout error, that's why I tried to wait for ajax and also i dont want to increase capybara timeout error just for one feature testing.

( tbh, I don't have much experience with capybara, so was just trying to fiddle around 😅)

Reply:
That's a lot of code for changing something that probably depends on the hardware the test runs on. I would rather read `Capybara.default_wait_time` from an environment variable so you can set this individually and we can leave the default as it is. Please do it this way :-)

Reply:
actually, I think instead of changing capybara's default wait time we should use wait as :

    expect(page).to have_selector(:css, "em", text: 'italic', wait: 7)

because capybara will wait for every test whether some element does exist or not.
(please do tell me if I m wrong somewhere 😅 )

However, I think there's some other problem since none of these works, even after waiting or  setting the time to 30 seconds! , that's why I tried all those lines of code but even those seem to pass only sometimes ...😕

Reply:
Try just increasing the `Capybara.default_wait_time` in the helper, see if that helps.

Reply:
already did that, no success.